### ISSUE TITLE
Error for "swish" activation function

### ISSUE THREAD

**Issue by agirbau** [AlexeyAB/darknet#4278]
Hello!

I was playing with new parameters and changed a model to use swish activations instead of leaky ones. When training it raises this error:

CUDA status Error: file: ./src/dark_cuda.c : () : line: 317 : build time: Nov 11 2019 - 11:36:28 
CUDA Error: an illegal memory access was encountered
CUDA Error: an illegal memory access was encountered: Resource temporarily unavailable
darknet: ./src/utils.c:293: error: Assertion `0' failed.

**Comment by agirbau** [comment]
Doing a pull with the last change doesn't work either:

CUDA status Error: file: ./src/dark_cuda.c : () : line: 317 : build time: Nov 12 2019 - 11:09:09

**Comment by LukeAI** [comment]
can you upload the .cfg ? I haven't had any such problems with swish. Have you tried an identical cfg with leaky?

**Comment by agirbau** [comment]
An example of a block in the cfg:

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=swish

[shortcut]
from=-3
activation=swish

Another network with exactly the same parameters but with activation=leaky does not crash. Maybe it's a CUDA version thing or something?

**Comment by kossolax** [comment]
Are you using a pretrained file from ReLu to retrain with Swish ?

**Comment by LukeAI** [comment]
> Are you using a pretrained file from ReLu to retrain with Swish ?
That shouldn't matter, I get better results using pretrained leaky-relu weights as a starting point for a swish network.

**Comment by LukeAI** [comment]
> An example of a block in the cfg:
> 
> [convolutional]
> batch_normalize=1
> filters=64
> size=3
> stride=1
> pad=1
> activation=swish
> 
> [convolutional]
> batch_normalize=1
> filters=32
> size=3
> stride=1
> pad=1
> activation=swish
> 
> [convolutional]
> batch_normalize=1
> filters=64
> size=3
> stride=1
> pad=1
> activation=swish
> 
> [shortcut]
> from=-3
> activation=swish
> 
> Another network with exactly the same parameters but with activation=leaky does not crash. Maybe it's a CUDA version thing or something?

I dunno but I doubt it to be honest. you've reproduced this error on two different versions of this repo? can you upload the whole cfg.

**Comment by AlexeyAB** [comment]
I fixed swish and mish activations for [shortcut] layer

**Comment by agirbau** [comment]
Thank you! Closing the issue!

**Comment by agirbau** [comment]
Hello!

I pulled and tested it for both swish and mish activation functions and got the same problem:

 CUDA allocate done! 
Loaded: 0.037322 seconds
CUDA status Error: file: ./src/dark_cuda.c : () : line: 317 : build time: Nov 12 2019 - 15:15:50 
CUDA Error: an illegal memory access was encountered
CUDA Error: an illegal memory access was encountered: Resource temporarily unavailable
darknet: ./src/utils.c:293: error: Assertion `0' failed.

**Comment by AlexeyAB** [comment]
* attach your cfg-file
* show your command
* show screenshot of the error

**Comment by agirbau** [comment]
cfg file:

[net]
# Testing
#batch=1
#subdivisions=1
# Training
batch=16
subdivisions=8
width=960
height=544
#width=1280
#height=736
#width=1440
#height=800

channels=3
momentum=0.9
decay=0.0005
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 50200
policy=steps
steps=20000,35000
scales=.1,.1

adam=1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=swish

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=swish

[shortcut]
from=-3
activation=swish

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=swish

[shortcut]
from=-3
activation=swish

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=swish

[shortcut]
from=-3
activation=swish

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=swish

[convolutional]
size=1
stride=1
pad=1
filters=44 
activation=linear

[Gaussian_yolo]
mask = 0,1,3,4
anchors = 5,5, 9,9, 15,30, 24,41, 34,59
classes=2
num=5
max = 30

jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
focal_loss=1



[route]
layers = -3

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=swish

[upsample]
stride=2

[route]
layers = -1, 12

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=swish

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=swish

[convolutional]
size=1
stride=1
pad=1
filters=44  
activation=linear

[Gaussian_yolo]
mask = 0,1,2,3
anchors = 5,5, 9,9, 15,30, 24,41, 34,59
classes=2
num=5
max=30

jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
focal_loss=1
